src/reconcile/rec_visibility.c

```diff
@@ -267,13 +266,12 @@ __wt_rec_upd_select(WT_SESSION_IMPL *session, WT_RECONCILE *r, WT_INSERT *ins, v
 
         /*
          * FIXME-prepare-support: A temporary solution for not storing durable timestamp in the
-         * cell. Properly fix this problem in PM-1524. It is currently not OK to write prepared
-         * updates with durable timestamp larger than checkpoint timestamp to data store as we don't
-         * store durable timestamp in the cell. However, it is OK to write them to the history store
-         * as we store the durable timestamp in the history store value.
+         * cell. Properly fix this problem in PM-1524. Currently pin all the prepared updates with
+         * durable timestamp larger than stable timestamp in cache.
          */
-        if (upd->durable_ts != upd->start_ts && upd->durable_ts > checkpoint_timestamp) {
+        if (upd->durable_ts != upd->start_ts && upd->durable_ts > r->stable_ts) {
```

Comment:
Is this right? Do we need to keep all updates with durable timestamps (that differ from their commit timestamp) more recent than the checkpoint serviced by the last completed checkpoint? Or is it enough that it *should* be flushed in the next checkpoint?

Reply:
Sorry, I don't follow the question.

Reply:
The stable timestamp tracks what MongoDB says is the most recent time it's safe to take a checkpoint at. I'm not sure if we need that timestamp here, or the timestamp of the most recently completed (or running) checkpoint.

Reply:
All we're trying to do is make sure we don't write any updates where the `durable_timestamp` is important, because at the moment we lose that information when writing the update to the page.  Here "important" is the tricky test -- once they're greater than the stable timestamp, they can't be rolled back in the live session, but they also can't be included in a checkpoint with a lower checkpoint timestamp (or recovery can't start from a state matching a point in the oplog).